static/extensions/Ikelene/email2FA.js

```diff
@@ -0,0 +1,317 @@
+                        blockType: Scratch.BlockType.BOOLEAN,
+                        text: 'Is 2FA server online?'
+                    },
+                    {
+                        blockType: Scratch.BlockType.LABEL,
+                        text: 'Customization'
+                    },
+                    {
+                        opcode: 'setEmailPreset',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'Set email style preset to [PRESET]',
+                        arguments: {
+                            PRESET: {
+                                type: Scratch.ArgumentType.STRING,
+                                menu: 'presetMenu'
+                            }
+                        }
+                    },
+                    {
+                        opcode: 'setCustomHTML',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'Set custom email HTML to [HTML]',
+                        arguments: {
+                            HTML: {
+                                type: Scratch.ArgumentType.STRING,
+                                defaultValue: '<h1>Your Code</h1>'
+                            }
+                        }
+                    },
+                    {
+                        blockType: Scratch.BlockType.LABEL,
+                        text: 'Use your own email'
+                    },
+                    {
+                        opcode: 'setCustomEmailLogin',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'Use custom email login [EMAIL] pass [PASS] smtp [SMTP] port [PORT]',
+                        arguments: {
+                            EMAIL: { type: Scratch.ArgumentType.STRING, defaultValue: 'your@email.com' },
+                            PASS: { type: Scratch.ArgumentType.STRING, defaultValue: 'password' },
+                            SMTP: { type: Scratch.ArgumentType.STRING, defaultValue: 'smtp.gmail.com' },
+                            PORT: { type: Scratch.ArgumentType.STRING, defaultValue: '465' }
+                        }
+                    },
+                    {
+                        opcode: 'clearCustomEmailLogin',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'Clear custom email login'
+                    }
+                ],
+                menus: {
+                    presetMenu: {
+                        acceptReporters: true,
+                        items: [
+                            { text: 'Classic (default)', value: '5' },
+ { text: 'Yellow Card', value: '1' },
+ { text: 'Blue Card', value: '2' },
+ { text: 'Minimalist Dashed', value: '3' },
+ { text: 'Welcome Style', value: '4' }
+                        ]
+                    }
+                }
+            };
+        }
+
+        constructor() {
+            const runtime = Scratch.runtime;
```

Comment:
The runtime is literally passed to the constructor as the first argument. Save the runtime under the current object rather than grabbing a reference to the runtime every time. Like,
`this.runtime = runtime`.